LykkeApi2/Controllers/OrdersController.cs

```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
             }
 
             return Created(Url.Action("Get", "Operations", new { id }), id);                  
-        }        
+        }
+        
+        [HttpDelete("limit/{orderId}")]
+        [SwaggerOperation("CancelLimitOrderNew")]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> CancelLimitOrderNew(string orderId)
+        {
+            var tradingSession = await _clientSessionsClient.GetTradingSession(_lykkePrincipal.GetToken());
+
+            var confirmationRequired = _baseSettings.EnableSessionValidation && !(tradingSession?.Confirmed ?? false);
+            if (confirmationRequired)
+            {
+                return BadRequest("Session confirmation is required");
+            }
+
+            var clientId = _requestContext.ClientId;
+
+            var activeOrders = await _limitOrdersRepository.GetActiveByClientIdAsync(clientId);
+
+            if (activeOrders.All(x => x.Id != orderId))
+                return NotFound();
+            
+            await _limitOrdersRepository.CancelByIdAsync(clientId, orderId);
+            await _matchingEngineClient.CancelLimitOrderAsync(orderId);
+            
+            return Ok();
+        }
+
+        [HttpDelete("limit")]
+        [SwaggerOperation("CancelMultipleLimitOrders")]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> CancelMultipleLimitOrders([FromBody]LimitOrderCancelMultipleRequest model)
+        {
+            var tradingSession = await _clientSessionsClient.GetTradingSession(_lykkePrincipal.GetToken());
+
+            var confirmationRequired = _baseSettings.EnableSessionValidation && !(tradingSession?.Confirmed ?? false);
+            if (confirmationRequired)
+            {
+                return BadRequest("Session confirmation is required");
+            }
+
+            if (!string.IsNullOrEmpty(model.AssetPairId))
+            {
+                var pair = await _assetsServiceWithCache.TryGetAssetPairAsync(model.AssetPairId);
+                
+                if (pair == null)
+                {
+                    return NotFound($"Asset pair '{model.AssetPairId}' not found.");
+                }
+            }
+
+            var clientId = _requestContext.ClientId;
+            
+            var activeOrders = await _limitOrdersRepository.GetActiveByClientIdAsync(clientId);
+
+            if (!activeOrders.Any())
+                return NotFound();
+
+            await _limitOrdersRepository.CancelMultipleAsync(
```

Comment:
Weird code. There can be inconsistency problems. But if it is an approved architecture decision, ok.

Reply:
Yes, there is no need to call it here. TxHandler will do the same things after ME executes the command.